packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -383,4 +388,55 @@ export class ProjectModel {
             })
             .where('project_uuid', projectUuid);
     }
+
+    async getCacheExplores(
+        projectUuid: string,
+    ): Promise<(Explore | ExploreError)[]> {
+        const explores = await this.database(CacheExploresTableName)
+            .select(['explores'])
+            .where('project_uuid', projectUuid)
+            .limit(1);
+        if (explores.length > 0) return explores[0].explores;
+        return [];
+    }
+
+    async saveCacheExplores(
+        projectUuid: string,
+        explores: (Explore | ExploreError)[],
+    ): Promise<void> {
+        this.database.transaction(async (trx) => {
+            await trx(CacheExploresTableName)
+                .where('project_uuid', projectUuid)
+                .delete();
+
+            const result = await trx(CacheExploresTableName).insert({
+                project_uuid: projectUuid,
+                explores: JSON.stringify(explores),
+            });
+        });
+    }
+
+    async lockProcess(projectUuid: string, func: () => void): Promise<void> {
+        this.database.transaction(async (trx) => {
+            // pg_advisory_xact_lock takes a 64bit integer as key
+            // we can't use project_uuid (uuidv4) as key, not even a hash,
+            // so we will be using autoinc project_id from DB.
+            const projects = await this.database('projects')
```

Comment:
You don't need the roundtrip here to the database, get the lock value by filtering the `projects` table on uuid:

```sql
SELECT
  pg_try_advisory_xact_lock('1', project_id)
FROM
  projects
WHERE
  project_uuid = ${project_uuid}
LIMIT 1
```

Reply:
Is there any way to avoid the "quoting" when using select ? 

```
            const projects = await this.database('projects')
                .where('project_uuid', projectUuid)
                .select(`pg_try_advisory_xact_lock(${CACHED_EXPLORES_PG_LOCK_NAMESPACE}, project_id)`)
                .limit(1);
```

2022-04-22 09:43:32 [Lightdash] error: Unhandled Rejection at Promise select "pg_try_advisory_xact_lock(1, project_id)" from "projects" where "project_uuid" = $1 limit $2 - column "pg_try_advisory_xact_lock(1, project_id)" does not exist

Unless I make everything `raw`

Reply:
Yeah I'm not sure how well `Knex` can handle this, it might be better as raw!